### PR TITLE
show message on sending empty log (fix #13476)

### LINKED ISSUE #13476: Trying to send empty log results in "unknown error"
### Describe your problem!

From a user report and reproducible also on `master`:

When trying to send a log with an empty log text c:geo reports an unknown error.

### How to reproduce?

- Open geocache
- Start log activity
- Make sure the log text is totally empty
- Try to submit the log

### Actual result after these steps?

c:geo reports an unknown error and offers to retry/store/drop.

Debug log contains:
[AsyncTask #2] GCWebAPI.postLog: No log text given {GCWebAPI.postLog:1004/GCLoggingManager.postLog:174/LogCacheActivity$Poster.doInBackgroundInternal:719/LogCacheActivity$Poster.doInBackgroundInternal:701/AbstractAsyncTaskWithProgress.doInBackground:122/AsyncTask$3.call:394/FutureTask.run:264/AsyncTask$SerialExecutor$1.run:305}

### Expected result after these steps?

c:geo should provide clear indication of the error reason or (better) not even try submitting but check this condition in beforehand and notify the user accordingly.

IMHO we had such a feature before, that blocked the sending if the text was empty....but I might be wrong.

### Reproducible

Yes

### c:geo Version

2022.09.04

### System information

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by murggel**
## Description

1. check length before posting log and show toast, if log is empty ("c:geo couldn\'t submit your log cause it is empty. Please be so kind and leave some nice words to the owner.")
2. show reason for failing posting log
3. check empty log on sending post request and set logResult accordingly

## Related issues

fix #13476

## Additional context

to discuss - if 2) and 3) is sufficient or if 1) is the better implementation

### CODE REVIEW

**Comment by murggel** [comment]
I will provide screenshots for the variants...

**Comment by murggel** [comment]
|Variant 1|Variant 2|
|----|-----|
|commit 1|commit 2 & 3|
|ShowToast_short  |PostingLogFailed|

**Comment by murggel** [comment]
With current implementation variant 1 will win.

Please comment, which variant should be implemented.

**Comment by eddiemuc** [comment]
I vote for variant 1

**Comment by Lineflyer** [comment]
variant 1

**Comment by moving-bits** [comment]
variant 1 should be fine

**Comment by murggel** [comment]
ready for review

File: main/src/main/java/cgeo/geocaching/log/LogCacheActivity.java [hunk 1]
```diff
@@ -513,10 +513,15 @@ public boolean onOptionsItemSelected(final MenuItem item) {
...
+            final int logLength = TextUtils.getNormalizedStringLength(binding.log.getText().toString());
+            if (logLength > 0) {
+                if (logLength <= LOG_MAX_LENGTH) {
+                    sendLogAndConfirm();
+                } else {
+                    Toast.makeText(this, R.string.cache_log_too_long, Toast.LENGTH_LONG).show();
+                }
             } else {
-                Toast.makeText(this, R.string.cache_log_too_long, Toast.LENGTH_LONG).show();
+                Toast.makeText(this, R.string.cache_empty_long, Toast.LENGTH_LONG).show();
```
Comment (moving-bits) [inline main/src/main/java/cgeo/geocaching/log/LogCacheActivity.java:524]: Just a tiny issue: Maybe rename string resource to "cache_empty_log"? (leaving out the "n")

Comment (murggel) [reply main/src/main/java/cgeo/geocaching/log/LogCacheActivity.java:524]: Correct :-)

**Pull request merged**